xml/System.Diagnostics/DiagnosticListener.xml

```diff
@@ -311,9 +316,11 @@
         <Parameter Name="isEnabled" Type="System.Predicate&lt;System.String&gt;" />
       </Parameters>
       <Docs>
-        <param name="observer">To be added.</param>
-        <param name="isEnabled">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="observer">An <see cref="T:System.IObserver" /> subscriber.</param>
+        <param name="isEnabled">Filters events based on their name (<see cref="T:System.String" />). Should return <see langword="true" /> if the event is enabled.  
+Note that the <paramref name="isEnabled" /> predicate is an optional optimization to allow the instrumentation site to avoid setting up the payload and calling <see cref="M:System.Diagnostics.DiagnosticListener.Write(System.String,System.Object)" /> when no subscriber cares about it. In particular the instrumentation site has the option of ignoring the <see cref="M:System.Diagnostics.DiagnosticListener.IsEnabled" /> predicate (not calling it) and simply calling <see cref="M:System.Diagnostics.DiagnosticListener.Write(System.String,System.Object)" />.   Thus if the subscriber requires the filtering, it needs to do it itself.
```

Comment:
This should be in the Remarks section, not the parameter description.

Reply:
Will edit these after I merge your suggestions.